go.mod

```diff
@@ -1,7 +1,7 @@
 // gomodjail:confined
 module github.com/lima-vm/lima
 
-go 1.23.0
+go 1.24
```

Comment:
Not sure if that is acceptable for us. It looks like a gratuitous bump; should I ask upstream why they changed it?

Reply:
Let's keep 1.23 unless this update is urgent

Reply:
Revert in #3535

Reply:
And as far as I know, you even have to depend on "1.24.0" now (since go1.21). None of this fuzzy/friendly

Reply:
I've now taken a closer look, and the requirement for 1.24 in `yq` does come from one of its dependencies: https://github.com/a8m/envsubst/pull/66

This still looks like a misguided use of the `go` line to force the user to upgrade Go, but I don't think there is any point in following up on this.

I think we should stay with 1.23 until Lima 1.10 is out (and keep `yq` at 4.45.1 until then), and then consider moving to Go 1.24 afterwards.

Reply:
> This still looks like a misguided use

You can see the misunderstanding by how in CI they (`a8m/envsubst`) still test compiling with both `1.23` and `1.24`.

This makes no sense because the `go 1.24` line will ensure that both jobs are compiled with `1.24`.